### ISSUE TITLE
Error displaying dask array with NAPARI_ASYNC and NAPARI_OCTREE

### ISSUE THREAD

**Issue by d-v-b** [napari/napari#2230]
This code block:

```python
import napari
import os
import dask.array as da

for envvar in ('NAPARI_ASYNC', 'NAPARI_OCTREE'):
    print(f'{envvar} : {os.environ.get(envvar)}')

ndim = 2
data = da.random.randint(0,255, (2048,) * ndim, chunks=(64,) * ndim, dtype='uint8')

with napari.gui_qt():
    viewer = napari.Viewer()
    viewer.add_image(data)

```

produces this traceback:
```bash
NAPARI_ASYNC : 1
NAPARI_OCTREE : 1
ERROR:root:Unhandled exception:
Traceback (most recent call last):
  File "/Users/bennettd/dev/napari/napari/_qt/qt_event_loop.py", line 213, in gui_qt
    yield app
  File "/Users/bennettd/dev/napari/examples/test.py", line 14, in <module>
    viewer.add_image(data)
  File "/Users/bennettd/dev/napari/napari/components/viewer_model.py", line 676, in add_image
    layer = image_class(data, **kwargs)
  File "/Users/bennettd/dev/napari/napari/layers/image/experimental/octree_image.py", line 75, in __init__
    super().__init__(*args, **kwargs)
  File "/Users/bennettd/dev/napari/napari/layers/image/image.py", line 276, in __init__
    self._update_dims()
  File "/Users/bennettd/dev/napari/napari/layers/base/base.py", line 530, in _update_dims
    self.refresh()
  File "/Users/bennettd/dev/napari/napari/layers/base/base.py", line 940, in refresh
    self.set_view_slice()
  File "/Users/bennettd/dev/napari/napari/layers/base/base.py", line 800, in set_view_slice
    self._set_view_slice()
  File "/Users/bennettd/dev/napari/napari/layers/image/experimental/octree_image.py", line 420, in _set_view_slice
    base_shape_2d = [base_shape[i] for i in self._dims_displayed]
  File "/Users/bennettd/dev/napari/napari/layers/image/experimental/octree_image.py", line 420, in <listcomp>
    base_shape_2d = [base_shape[i] for i in self._dims_displayed]
IndexError: tuple index out of range
```

If this errors, what kind of chunked images was the async functionality tested on?

**Comment by sofroniewn** [comment]
Hi @d-v-b thanks for filing this! As discussed in Zulip, i'll look into this more and get back to you!!

**Comment by sofroniewn** [comment]
Ah @d-v-b this errors because right now with `NAPARI_OCTREE=1` we're expecting a precomputed pyramid, sorry for missing that first time i looked at it.

Something like 

```python
ndim = 2
data = [da.random.randint(0, 255, (2**i_,) * ndim, chunks=(64,) * ndim) / 255 for i_ in range(12, 6, -1)]
``` 

would be better. I'm working on some other bug fixes in #2349 though, so you might still want to wait a little before trying this out!

**Comment by sofroniewn** [comment]
@d-v-b this should now be fixed on master - you can give a chunked single scale image and we'll load tiles as best we can! Very curious to hear about how it works for you.
